### PR TITLE
Fix some more linting errors.

### PR DESCRIPTION
**Pull request by tomprince**

### CODE REVIEW

File: .luacheckrc [hunk 1]
```diff
@@ -37,14 +37,11 @@ allow_defined_top = true
 ignore = {
     "131", -- unused global variable, we should annotate the ones we expect
     "542", -- empty if branch  (usually has a comment)
+    "212", -- unused argument; name is used as documentation
```
Comment (iakona) [inline .luacheckrc:40]: hmm i'm not opposed to having this enforced. feels a bit strange to have unused named arguments to me

File: objects/029995/script.lua [hunk 2]
```diff
@@ -74,11 +74,11 @@ function toggleObjects()
         end
         empty = false
     else
-        for _,obj in pairs(self.getObjects()) do
+        for _, bagObject in pairs(self.getObjects()) do
```
Comment (iakona) [inline objects/029995/script.lua:77]: what's up with all of these swaps?

Comment (tomprince) [reply objects/029995/script.lua:77]: There is a `function(obj) .... end` a few lines below that shadows this `obj`.

File: .luacheckrc (hunk 1 above)
Comment (tomprince) [reply .luacheckrc:40]: There are a bunch of places like [this](https://github.com/spirit-island/spirit-island-mod/blob/beta/global-script.lua#L283-85) where there are a bunch of parameter required as part of an API, and I think it is better to name the arguments, then to need to use `_` for all the unused arguments.
```
    addHotkey("Add Fear", function (droppingPlayerColor, hoveredObject, cursorLocation, key_down_up)
        aidBoard.call("addFear")
    end)
```
If it wasn't for that, I'd agree with you.

**Review by iakona — APPROVED** [review]

**Pull request merged**